fvm/transactionInvoker.go

```diff
@@ -296,3 +201,106 @@ func (i TransactionInvoker) normalExecution(
 
 	return
 }
+
+// Clear changes and try to deduct fees again.
+func (i TransactionInvoker) errorExecution(
+	proc *TransactionProcedure,
+	txnState *state.TransactionState,
+	nestedTxnId state.NestedTransactionId,
+	env environment.Environment,
+	errs *errors.ErrorsCollector,
+) error {
+	txnState.DisableAllLimitEnforcements()
+	defer txnState.EnableAllLimitEnforcements()
+
+	// log transaction as failed
+	env.Logger().Info().
+		Err(errs.ErrorOrNil()).
+		Msg("transaction executed with error")
+
+	env.Reset()
+
+	// drop delta since transaction failed
+	restartErr := txnState.RestartNestedTransaction(nestedTxnId)
+	if errs.Collect(restartErr) {
```

Comment:
note the minor change here.  This used to be 

```
		if restartErr != nil {
			return fmt.Errorf(
				"cannot restart nested transaction: %w "+
					"(original processErr: %v)",
				restartErr,
				processErr)
		}
```

They are functionally identical